File: 3/python/solution3.py
def largest_prime_factor(n):
    factor = 2
    while factor * factor <= n:
        if n % factor == 0:
            n //= factor  
        else:
            factor += 1  
    return n 

number = 600851475143
result = largest_prime_factor(number)
print("The largest prime factor of", number, "is", result)

File: 3/go/solution3.go
package main

import "fmt"

func main() {
	number := 600851475143
	factor := 2

	for factor*factor <= number {
		if number%factor == 0 {
			number /= factor // Divide number by factor
		} else {
			factor += 1 // Move to the next factor
		}
	}

	fmt.Println("The largest prime factor is:", number)
}
